ssl/quic/quic_demux.c

```diff
@@ -487,13 +499,35 @@ static QUIC_DEMUX_CONN *demux_identify_conn(QUIC_DEMUX *demux, QUIC_URXE *e)
 static int demux_process_pending_urxe(QUIC_DEMUX *demux, QUIC_URXE *e)
 {
     QUIC_DEMUX_CONN *conn;
+    int r;
 
     /* The next URXE we process should be at the head of the pending list. */
     if (!ossl_assert(e == ossl_list_urxe_head(&demux->urx_pending)))
         return 0;
 
     assert(e->demux_state == URXE_DEMUX_STATE_PENDING);
 
+    /*
+     * Check if the packet ends with a stateless reset token and if it does
+     * skip it after dropping the connection.
+     *
+     * RFC 9000 s. 10.3.1 Detecting a Stateless Reset
+     *      If the last 16 bytes of the datagram are identical in value to
+     *      a stateless reset token, the endpoint MUST enter the draining
+     *      period and not send any further packets on this connection.
+     *
+     * Returning a failure here causes the connection to enter the terminating
+     * state which achieves the desired outcome.
+     */
```

Comment:
There is definitely some TODO(QUIC SERVER) and TODO(QUIC FUTURE) here. One thing is that we do not need to try to match all the packets - it is sufficient to check for the stateless reset token only if we are unable to identify the connection and decrypt the packet.

Reply:
Agreed about the not needing to check everything, I'll tag that future.

What's server here?  The same rules apply to both parties in a connection.